wrapper/Ada/tls_client.adb

```diff
@@ -40,8 +41,62 @@ package body Tls_Client with SPARK_Mode is
 
    subtype Byte_Type is WolfSSL.Byte_Type;
 
+   subtype chars_ptr is WolfSSL.chars_ptr;
+   subtype unsigned is WolfSSL.unsigned;
+
    package Natural_IO is new Ada.Text_IO.Integer_IO (Natural);
 
+   function PSK_Client_Callback
+     (Unused         : WolfSSL.WolfSSL_Type;
+      Hint           : chars_ptr;
+      Identity       : chars_ptr;
+      Id_Max_Length  : unsigned;
+      Key            : chars_ptr;
+      Key_Max_Length : unsigned) return unsigned
+   with Convention => C;
+
+   function PSK_Client_Callback
+     (Unused         : WolfSSL.WolfSSL_Type;
+      Hint           : chars_ptr;
+      Identity       : chars_ptr;
+      Id_Max_Length  : unsigned;
+      Key            : chars_ptr;
+      Key_Max_Length : unsigned) return unsigned
+   with
+     SPARK_Mode => Off      
+   is
+      use type Interfaces.C.unsigned;
+
+      Hint_String : constant String := Interfaces.C.Strings.Value (Hint);
+      Identity_String : constant String := "Client_identity";
+      Key_String : constant String :=
```

Comment:
This is just a sample test key from https://github.com/wolfSSL/wolfssl/blob/master/wolfssl/test.h#L1909
`/* test key in hex is 0x1a2b3c4d , in decimal 439,041,101 , we're using unsigned binary */`

Reply:
As said, my inspiration (and test counterpart) was https://github.com/wolfSSL/wolfssl-examples/blob/master/psk/server-psk.c

I can change the values to hexadecimal as in `test.h` if that is what you are suggesting.

Reply:
You don't need to change any code. Please add inline code comment referencing why these test values were chosen.